model/rag/chat.go

```diff
@@ -85,9 +85,10 @@ type Source struct {
 	ChunkURL     string `json:"chunkUrl"`
 	Page         int    `json:"page"`
 	EmailPreview string `json:"email.preview,omitempty"`
-	GroupID      string `json:"email.groupId,omitempty"`
+	GroupId      string `json:"group_id,omitempty"`
+	ParentId     string `json:"parent_id,omitempty"`
```

Comment:
nit: We usually write id with uppercases in our variable and attribute names so it might be good to stay consistent with the rest of the code base and use `GroupID` and `ParentID` here.